source/contributed/tutorialMonkeyPatching.md

```diff
@@ -0,0 +1,210 @@
+# Monkey Patching Tutorial for Screeps
+WARNING: Monkey patching is often considered bad practice and there are those that will tell you that you should NEVER do it, and there are good reasons for this. Monkey patching, while very powerful and convenient (especially in Screeps), is a dangerous practice that should not be taken lightly.
+
+Monkey patching is the practice of modifying the way that existing prototype functions operate.
+
+We will provide examples that will demonstrate a few of the infinite uses for monkey patching. Each example will build on the previous one, so you'll want to read through all of it to make sure you understand it fully.
+
+
+## Introduction - Adding functions to prototypes
+
+To understand monkey patching it is important to understand that functions are/can be stored in variables just like numbers, objects, arrays, and strings. Creep functions like `.moveTo`, `.harvest`, or `.withdraw` are all stored in variables belonging to the Creep prototype object. These functions can be accessed like so: `Creep.prototype.moveTo`, `Creep.prototype.harvest`, and `Creep.prototype.withdraw`. Any values that belong to the Creep prototype will be accessible from any Creep object.  This is why you are able to call `creep.moveTo` on all of your creeps. 
```

Comment:
Please format all API objects and methods mentions as links to the API reference. This also includes JavaScript language features as well (you can use [Mozilla MDN Reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript) for this).